### PR TITLE
Add deneb config for Chiado

### LINKED ISSUE #7871: Schedule Chiado testnet Deneb fork (expected on Jan, 31)
### Description
Chiado network will be transitioned to Deneb hardfork on January, 31 2024
https://github.com/gnosischain/configs/pull/26
Target blobs per block is 1, so it's expected that required blobs will take up about 35G of storage

Gnosis and Chiado deneb presets: https://github.com/gnosischain/specs/blob/master/consensus/preset/gnosis/deneb.yaml
Deneb config lines: https://github.com/gnosischain/specs/blob/master/consensus/config/gnosis.yaml#L128-L136
Chiado config lines: https://github.com/gnosischain/configs/pull/27
Trusted setup: https://github.com/gnosischain/specs/pull/30

### PR DESCRIPTION
**Pull request by rolfyone**
fixes #7871

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by rolfyone** [comment]
Theres probably a few things to push up stream so that there's less changes in this between the published and our version...

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml [hunk 1]
```diff
@@ -156,10 +156,15 @@ BELLATRIX_FORK_EPOCH: 180 # Mon Oct 10 2022 14:00:00 GMT+0000
...
 CAPELLA_FORK_EPOCH: 244224 # Wed May 24 2023 13:12:00 GMT+0000
+# Deneb
+DENEB_FORK_VERSION: 0x0400006f
+DENEB_FORK_EPOCH: 516608 # Wed Jan 31 2024 18:15:40 GMT+0000
 
 INACTIVITY_SCORE_BIAS: 4
 # 2**4 (= 16)
 INACTIVITY_SCORE_RECOVERY_RATE: 16
+# [New in Deneb:EIP7514] 2**3 (= 8)
+MAX_PER_EPOCH_ACTIVATION_CHURN_LIMIT: 8
```
Comment (zilm13) [inline ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: I'm not sure on this. It's set it to [2 for gnosis](https://github.com/gnosischain/specs/blob/master/consensus/config/gnosis.yaml#L77-L78) but I don't see a line in chiado config.

**Review by zilm13 — COMMENTED** [review]
You probably need to add a line like this too https://github.com/ConsenSys/teku/blob/5ac9458e040f900cf61d164ff8084176239d22db/ethereum/networks/src/main/java/tech/pegasys/teku/networks/Eth2NetworkConfiguration.java#L653

Also we could update gnosis deneb settings, but it could be done in separate PR

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml (hunk 1 above)
Comment (rolfyone) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: oh is it? I couldnt find it and it was needed, so i copied from mainnet... if theres a gnosis one we can copy that, but ideally it should be in their actual config :/ same as fork_choice settings which were missing

**Comment by rolfyone** [comment]
Ultimately what i want is to be able to have https://github.com/gnosischain/configs/blob/main/chiado/config.yaml in our chiado.yaml, but its very different currently...
(below the `chiado.yaml` is the one from the gnosis ref above)

```
diff chiado.yaml ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml
41a42,51
> # Fork Choice
> # ---------------------------------------------------------------
> # 2**3 (= 8)
> SAFE_SLOTS_TO_UPDATE_JUSTIFIED: 8
> # 20%
> REORG_HEAD_WEIGHT_THRESHOLD: 20
> # 160%
> REORG_PARENT_WEIGHT_THRESHOLD: 160
> # `2` epochs
> REORG_MAX_EPOCHS_SINCE_FINALIZATION: 2
48,49d57
< # 2**0 (= 1)
< RANDOM_SUBNETS_PER_VALIDATOR: 1
157a166,167
> # [New in Deneb:EIP7514] 2**3 (= 8)
> MAX_PER_EPOCH_ACTIVATION_CHURN_LIMIT: 8
158a169,197
> # Networking
> # ---------------------------------------------------------------
> # `10 * 2**20` (= 10485760, 10 MiB)
> GOSSIP_MAX_SIZE: 10485760
> # `2**10` (= 1024)
> MAX_REQUEST_BLOCKS: 1024
> # `2**8` (= 256)
> EPOCHS_PER_SUBNET_SUBSCRIPTION: 256
> ## `MIN_VALIDATOR_WITHDRAWABILITY_DELAY + CHURN_LIMIT_QUOTIENT // 2`
> MIN_EPOCHS_FOR_BLOCK_REQUESTS: 2304
> # `10 * 2**20` (=10485760, 10 MiB)
> MAX_CHUNK_SIZE: 10485760
> # 5s
> TTFB_TIMEOUT: 5
> # 10s
> RESP_TIMEOUT: 10
> ATTESTATION_PROPAGATION_SLOT_RANGE: 32
> # 500ms
> MAXIMUM_GOSSIP_CLOCK_DISPARITY: 500
> MESSAGE_DOMAIN_INVALID_SNAPPY: 0x00000000
> MESSAGE_DOMAIN_VALID_SNAPPY: 0x01000000
> # 2 subnets per node
> SUBNETS_PER_NODE: 2
> # 2**8 (= 64)
> ATTESTATION_SUBNET_COUNT: 64
> ATTESTATION_SUBNET_EXTRA_BITS: 0
> # ceillog2(ATTESTATION_SUBNET_COUNT) + ATTESTATION_SUBNET_EXTRA_BITS
> ATTESTATION_SUBNET_PREFIX_BITS: 6
```

 - The fork choice section was in our previous file, so i copied that back in,
 - `MAX_PER_EPOCH_ACTIVATION_CHURN_LIMIT` copied from mainnet which sounds like it was incorrect
 - networking section was in our previous implementation, so I copied that.
I'm not sure where these fields come from for everyone else, so it's a bit curious. I wonder if @daplion has any input on these...

It does look like I missed `RANDOM_SUBNETS_PER_VALIDATOR` from the diff above... I'll double check that...

To me the 2 missing sections, and probably MAX_PER_EPOCH_ACTIVATION_CHURN_LIMIT should all be in the config source we copy...

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml (hunk 1 above)
Comment (zilm13) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: click a link, it's an official gnosis config (as I was told)

**Comment by rolfyone** [comment]
just on RANDOM_SUBNETS_PER_VALIDATOR, that flags as an unknown option:
```
2024-01-16 06:48:19.441+10:00 | main | INFO  | SpecConfigReader | Ignoring non-configurable constants supplied in network configuration: BLS_WITHDRAWAL_PREFIX, TARGET_AGGREGATORS_PER_COMMITTEE, EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION, DOMAIN_BEACON_PROPOSER, DOMAIN_BEACON_ATTESTER, DOMAIN_RANDAO, DOMAIN_DEPOSIT, DOMAIN_VOLUNTARY_EXIT, DOMAIN_SELECTION_PROOF, DOMAIN_AGGREGATE_AND_PROOF, DOMAIN_SYNC_COMMITTEE, DOMAIN_CONTRIBUTION_AND_PROOF, DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF

java.lang.IllegalArgumentException: Unable to load configuration for network "chiado": Detected unknown spec config entries: RANDOM_SUBNETS_PER_VALIDATOR
```

This looks to be because: 
https://github.com/ethereum/consensus-specs/issues/2749 which starts using different network fields, and also explains a lot of the network fields...

**Comment by zilm13** [comment]
this option was outdated with spec changes see #7341

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml [hunk 2]
```diff
@@ -156,10 +156,15 @@ BELLATRIX_FORK_EPOCH: 180 # Mon Oct 10 2022 14:00:00 GMT+0000
...
 CAPELLA_FORK_VERSION: 0x0300006f
 CAPELLA_FORK_EPOCH: 244224 # Wed May 24 2023 13:12:00 GMT+0000
+# Deneb
+DENEB_FORK_VERSION: 0x0400006f
+DENEB_FORK_EPOCH: 516608 # Wed Jan 31 2024 18:15:40 GMT+0000
 
 INACTIVITY_SCORE_BIAS: 4
 # 2**4 (= 16)
 INACTIVITY_SCORE_RECOVERY_RATE: 16
+# [New in Deneb:EIP7514] 2**3 (= 8)
```
Comment (zilm13) [inline ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: 2 in comment too (very nit)

**Review by zilm13 — COMMENTED** [review]
You probably need to add a line like this too https://github.com/ConsenSys/teku/blob/5ac9458e040f900cf61d164ff8084176239d22db/ethereum/networks/src/main/java/tech/pegasys/teku/networks/Eth2NetworkConfiguration.java#L653

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml (hunk 2 above)
Comment (rolfyone) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: it was 8 in the comment in the gnosis :D

**Comment by rolfyone** [comment]
> You probably need to add a line like this too https://github.com/ConsenSys/teku/blob/5ac9458e040f900cf61d164ff8084176239d22db/ethereum/networks/src/main/java/tech/pegasys/teku/networks/Eth2NetworkConfiguration.java#L653

yes but now we have a test that will break ;)

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml (hunk 2 above)
Comment (zilm13) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: `2*` whatever it means
https://github.com/gnosischain/specs/blob/master/consensus/config/gnosis.yaml#L77

Comment (rolfyone) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: bad edit, i'll just make ours 2

**Comment by dapplion** [comment]
@rolfyone thanks for noticing, the chiado config was not updated to the latest set of keys. Now fixed with 
- https://github.com/gnosischain/configs/pull/28

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml [hunk 3]
```diff
@@ -169,8 +104,8 @@ GOSSIP_MAX_SIZE: 10485760
 MAX_REQUEST_BLOCKS: 1024
 # `2**8` (= 256)
 EPOCHS_PER_SUBNET_SUBSCRIPTION: 256
-## `MIN_VALIDATOR_WITHDRAWABILITY_DELAY + CHURN_LIMIT_QUOTIENT // 2`
-MIN_EPOCHS_FOR_BLOCK_REQUESTS: 2304
+# `MIN_VALIDATOR_WITHDRAWABILITY_DELAY + CHURN_LIMIT_QUOTIENT // 2` (= 33024, ~5 months)
```
Comment (4rgon4ut) [inline ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]:
```suggestion
# `MIN_VALIDATOR_WITHDRAWABILITY_DELAY + CHURN_LIMIT_QUOTIENT // 2` (= 33024, ~31 days)
```

**Review by StefanBratanov — APPROVED** [review]
LGTM

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml [hunk 4]
```diff
@@ -156,10 +43,58 @@ BELLATRIX_FORK_EPOCH: 180 # Mon Oct 10 2022 14:00:00 GMT+0000
...
+DENEB_FORK_EPOCH: 516608 # Wed Jan 31 2024 18:15:40 GMT+0000
+
+
+# Time parameters
+# ---------------------------------------------------------------
+# 5 seconds
+SECONDS_PER_SLOT: 5
+# 6 (estimate from xDai mainnet)
+SECONDS_PER_ETH1_BLOCK: 6
+# 2**8 (= 256) epochs ~27 hours
```
Comment (4rgon4ut) [inline ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]:
```suggestion
# 2**8 (= 256) epochs ~5.7 hours
```

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml [hunk 5]
```diff
@@ -156,10 +43,58 @@ BELLATRIX_FORK_EPOCH: 180 # Mon Oct 10 2022 14:00:00 GMT+0000
...
+MIN_VALIDATOR_WITHDRAWABILITY_DELAY: 256
+# 2**8 (= 256) epochs ~27 hours
```
Comment (4rgon4ut) [inline ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]:
```suggestion
# 2**8 (= 256) epochs ~5.7 hours
```

**Review by 4rgon4ut — APPROVED** [review]
LGTM. Added some minor suggestions according to recent changes in https://github.com/gnosischain/configs/pull/31

File: ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml (hunk 3 above)
Comment (rolfyone) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: possibly missing the point that this is copied and pasted from chiado config?

Comment (rolfyone) [reply ethereum/spec/src/main/resources/tech/pegasys/teku/spec/config/configs/chiado.yaml]: I can change this but its not goingto be right next time we copy it...

**Comment by rolfyone** [comment]
> LGTM. Added some minor suggestions according to recent changes in [gnosischain/configs#31](https://github.com/gnosischain/configs/pull/31)

oh awesome - that makes sense then... thanks!

**Pull request merged**